prow/prstatus/prstatus.go

```diff
@@ -45,31 +45,28 @@ const (
 
 type githubClient interface {
 	Query(context.Context, interface{}, map[string]interface{}) error
-}
-
-type githubRestfulClient interface {
-	GetUser(login string) (*gogithub.User, error)
-}
-
-type pullRequestRestfulClient struct {
-	*ghclient.Client
-}
-
-func (grc pullRequestRestfulClient) GetUser(login string) (*gogithub.User, error) {
-	return grc.Client.GetUser(login)
+	GetCombinedStatus(org, repo, ref string) (*github.CombinedStatus, error)
 }
 
 // PullRequestQueryHandler defines an interface that query handlers should implement.
 type PullRequestQueryHandler interface {
+	HeadContexts(ghc githubClient, pr PullRequest) ([]Context, error)
 	QueryPullRequests(context.Context, githubClient, string) ([]PullRequest, error)
+	GetUser(*ghclient.Client) (*gogithub.User, error)
 }
 
 // UserData represents data returned to client request to the endpoint. It has a flag that indicates
 // whether the user has logged in his github or not and list of open pull requests owned by the
 // user.
 type UserData struct {
-	Login        bool
-	PullRequests []PullRequest
+	Login                    bool
+	PullRequestsWithContexts []PullRequestWithContext
+}
+
+// PullRequestWithContext contains a pull request with its latest context.
+type PullRequestWithContext struct {
+	Contexts    []Context
+	PullRequest PullRequest
```

Comment:
You could embed `PullRequest` here

Reply:
Did you mean Contexts could be a field of PullRequest

Reply:
@stevekuznetsov poke